sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/_tests_/unit/application-event-code-during-enrolment-and-completed-utils.service.applicationEventCodeDuringEnrolmentAndCompleted.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { COEStatus } from "@sims/sims-db";
+import { ApplicationEventCode } from "../../../models/ier12-integration.model";
+import { ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService } from "../..";
+import { DisbursementValueService } from "@sims/integrations/services";
+import { createMock } from "@golevelup/ts-jest";
+
+describe("ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService-applicationEventCodeDuringEnrolmentAndCompleted", () => {
+  let applicationEventCodeDuringEnrolmentAndCompletedUtilsService: ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService;
+
+  beforeAll(() => {
+    const disbursementValueService = createMock<DisbursementValueService>();
+    applicationEventCodeDuringEnrolmentAndCompletedUtilsService =
+      new ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService(
+        disbursementValueService,
+      );
+  });
+
+  it(`Should return ${ApplicationEventCode.COER} when the coe status is ${COEStatus.required}.`, () => {
+    // Act
+    const applicationEventCode =
+      applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringEnrolmentAndCompleted(
+        COEStatus.required,
+      );
+    // Assert
+    expect(applicationEventCode).toBe(ApplicationEventCode.COER);
+  });
+
+  it(`Should return ${ApplicationEventCode.COED} when the coe status is ${COEStatus.declined}.`, () => {
+    // Act
+    const applicationEventCode =
+      applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringEnrolmentAndCompleted(
+        COEStatus.declined,
+      );
+    // Assert
+    expect(applicationEventCode).toBe(ApplicationEventCode.COED);
+  });
+
+  it(`Should throw an error when the coe status is other than ${COEStatus.required} or ${COEStatus.declined}.`, () => {
+    // Arrange, act and assert.
+    expect(() => {
+      applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringEnrolmentAndCompleted(
+        COEStatus.completed,
+      );
+    }).toThrow("Unexpected coe status.");
```

Comment:
IMO acronyms like COE, NOA, MSFAA, and all should always be uppercase along comments.